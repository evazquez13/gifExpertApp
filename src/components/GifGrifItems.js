import React from 'react';
import PropTypes from 'prop-types';

export const GifGrifItems = ({title,url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGrifItems.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired    
} 