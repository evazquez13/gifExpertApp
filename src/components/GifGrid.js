import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGrifItems } from './GifGrifItems';

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDownBig">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading</p>}
            
            <div className="card-grid animate__animated animate__flipInY">
                {
                    data.map( img => (<GifGrifItems
                                            key={img.id}
                                            {...img}/>))
                }
                
            </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired
}
