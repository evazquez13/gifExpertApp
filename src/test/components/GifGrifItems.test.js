import React from 'react';
import { shallow } from 'enzyme';
import { GifGrifItems } from '../../components/GifGrifItems';

describe('Pruebas al componente <GifGrifItems/>',() => {
    
    const titulo = 'Un titulo';
    const url = 'htts://localhost/prueba.jpg';
    const wrapper = shallow(<GifGrifItems title={titulo} url={url}/>);
    
    test('Debe mostar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe tener un parrafo con titulo', () =>{

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);

    });

    test('Debe tener la imagen igual al url y alt de los props', () =>{

        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.prop('alt')).toBe(titulo);

    });

    test('El div debe tener la clase animate__fadeIn', () => {

        const div = wrapper.find('div');

        expect(div.props().className.includes('animate__fadeIn')).toBe(true);
        
    });
    

});