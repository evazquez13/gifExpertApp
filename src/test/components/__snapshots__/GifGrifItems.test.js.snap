// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrifItems/> Debe mostar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="htts://localhost/prueba.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
